File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/connection/builder/live/specs/GivenANetworkExists/AndThUrlIsNotBuilt/WhenGettingTheLiveUrl.java
package com.lasthopesoftware.bluewater.client.connection.builder.live.specs.GivenANetworkExists.AndThUrlIsNotBuilt;

import android.net.NetworkInfo;
import com.lasthopesoftware.bluewater.client.connection.builder.live.LiveUrlProvider;
import com.lasthopesoftware.bluewater.client.connection.url.IUrlProvider;
import com.lasthopesoftware.bluewater.client.library.repository.Library;
import com.lasthopesoftware.bluewater.shared.promises.extensions.specs.FuturePromise;
import com.namehillsoftware.handoff.promises.Promise;
import org.junit.BeforeClass;
import org.junit.Test;

import java.util.concurrent.ExecutionException;

import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
import static org.mockito.Mockito.mock;

public class WhenGettingTheLiveUrl {

	private static IUrlProvider urlProvider;

	@BeforeClass
	public static void before() throws ExecutionException, InterruptedException {
		final LiveUrlProvider liveUrlProvider = new LiveUrlProvider(
			() -> mock(NetworkInfo.class),
			(library) -> Promise.empty());
		urlProvider = new FuturePromise<>(liveUrlProvider.promiseLiveUrl(new Library())).get();
	}

	@Test
	public void thenTheUrlIsCorrect() {
		assertThat(urlProvider).isNull();
	}
}

File: projectBlueWater/src/test/java/com/lasthopesoftware/bluewater/client/connection/builder/live/GivenANetworkExists/AndThUrlIsNotBuilt/WhenGettingTheLiveUrl.kt
package com.lasthopesoftware.bluewater.client.connection.builder.live.GivenANetworkExists.AndThUrlIsNotBuilt

import com.lasthopesoftware.bluewater.client.browsing.library.repository.LibraryId
import com.lasthopesoftware.bluewater.client.connection.builder.BuildUrlProviders
import com.lasthopesoftware.bluewater.client.connection.builder.live.LiveUrlProvider
import com.lasthopesoftware.bluewater.shared.promises.extensions.toExpiringFuture
import com.namehillsoftware.handoff.promises.Promise
import io.mockk.every
import io.mockk.mockk
import org.assertj.core.api.AssertionsForClassTypes.assertThat
import org.junit.jupiter.api.Test

class WhenGettingTheLiveUrl {

	private val urlProvider by lazy {
		val builder = mockk<BuildUrlProviders>()
		every { builder.promiseBuiltUrlProvider(any()) } returns Promise.empty()

		val liveUrlProvider = LiveUrlProvider(
			{ mockk() },
			builder
		)
		liveUrlProvider.promiseLiveUrl(LibraryId(54)).toExpiringFuture().get()
	}

	@Test
	fun `then the url is correct`() = assertThat(urlProvider).isNull()
}
